specification/compute/Gallery.Management/GalleryScriptVersion.tsp

```diff
@@ -0,0 +1,108 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+import "@typespec/openapi";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "./models.tsp";
+import "./GalleryScript.tsp";
+import "../Common.Management/operations.tsp";
+
+using TypeSpec.Rest;
+using Versioning;
+using Azure.ResourceManager;
+using TypeSpec.Http;
+
+namespace Microsoft.Compute {
+  /**
+   * Gallery script version name with semantic version pattern validation
+   */
+  @pattern("^[0-9]+\\.[0-9]+\\.[0-9]+$")
+  @doc("The name of the gallery Script Version to be created. Needs to follow semantic version name pattern: The allowed characters are digit and period. Digits must be within the range of a 32-bit integer. Format: <MajorVersion>.<MinorVersion>.<Patch>")
+  scalar GalleryScriptVersionName extends string;
+
+  /**
+   * The List Gallery Script version operation response.
+   */
+  @added(Versions.v2025_03_03)
+  model GalleryScriptVersionList is Azure.Core.Page<GalleryScriptVersion>;
+
+  /**
+   * Specifies information about the gallery Script Version that you want to create or update.
+   */
+  @added(Versions.v2025_03_03)
+  @parentResource(GalleryScript)
+  model GalleryScriptVersion
+    is Azure.ResourceManager.TrackedResource<GalleryScriptVersionProperties> {
+    @key("galleryScriptVersionName")
+    @segment("versions")
+    @path
+    name: GalleryScriptVersionName;
+  }
+
+  @armResourceOperations
+  interface GalleryScriptVersions {
+    /**
+     * Retrieves information about a gallery Script Version.
+     */
+    @added(Versions.v2025_03_03)
+    get is ComputeResourceRead<GalleryScriptVersion>;
+
+    /**
+     * Create or update a gallery Script Version. Script versions helps save different states/iterations of a Gallery script. They are referred in format: <MajorVersion>.<MinorVersion>.<Patch>
+     */
+    #suppress "@azure-tools/typespec-azure-resource-manager/arm-put-operation-response-codes" "For backward compatibility"
+    #suppress "@azure-tools/typespec-azure-resource-manager/no-response-body" "For backward compatibility"
+    @added(Versions.v2025_03_03)
+    createOrUpdate is ComputeResourceCreateOrReplaceAsync<
+      GalleryScriptVersion,
+      Response = ArmResourceUpdatedResponse<GalleryScriptVersion> | ArmResourceCreatedResponse<
+        GalleryScriptVersion,
+        ArmLroLocationHeader & Azure.Core.Foundations.RetryAfterHeader
```

Comment:
[https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#creating-or-replacing-using-put](https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#creating-or-replacing-using-put)

"Starting January 2025, new Resource Provider namespace implementations must include the Azure-AsyncOperation header in their async responses."

This and other Async PUT APIs